src/postprocess/dos.jl

```diff
@@ -68,65 +69,200 @@ function compute_ldos(scfres::NamedTuple; ε=scfres.εF, kwargs...)
+            proj_vectors[:,iproj] = structure_factor .* form_factors[ik][:, iproj] ./ sqrt(basis.model.unit_cell_volume)    
+        end
+
+        @assert size(proj_vectors, 2) == nprojs "Projection matrix size mismatch: $(size(proj_vectors)) != $nprojs"
+        # At this point proj_vectors is a matrix containing all orbital projectors from all atoms. 
+        #   What we want is to have them all orthogonal, to avoid double counting in the Hubbard U term contribution.
+        #   We use Lowdin orthogonalization to minimize the "identity loss" of individual orbital projectors after the orthogonalization
+        proj_vectors = ortho_lowdin(proj_vectors)  
+        
+        projectors[ik] = proj_vectors  
+    end
+
+    return (;projectors, labels)
+end
+
+"""
+Build the projection matrices projsk for all k-points at the same time.
 
+    projs[ik][iband, iproj] = projsk[iband, iproj] = |<ψnk, ϕilm>|^2
+
+ where ψnk is the wavefunction for band iband at k-point kpt,
+  and ϕilm is the atomic orbital for atom i, quantum numbers (n,l,m)
+
+    Input:
+     -> basis  : PlaneWaveBasis
+     -> ψ      : Wavefunction
+    Output:
+     -> projs  : Vector of matrices of projections
+     -> labels : NamedTuple containing iatom, species, n, l, m and orbital name for each projector
+Note: 'n' is not the principal quantum number, but the index of the radial wavefunction in the pseudopotential.
+          So for Si, the 3s orbital would have n=1, l=0, m=0, and the 3p orbital would have n=1, l=1, m=-1,0,1.
+"""
+function build_projections(basis::PlaneWaveBasis{T}, ψ;
+                            manifold  = nothing,
+                            positions = basis.model.positions           
+                          ) where {T}
+    projectors, labels = build_projectors(basis; manifold=manifold, positions=positions)
     projs = Vector{Matrix}(undef, length(basis.kpoints))
     for (ik, ψk) in enumerate(ψ)
-        structure_factor = [cis2pi(-dot(position, p)) for p in G_plus_k_all[ik]]
-        # TODO orthogonalize pseudo-atomic wave functions?
-        proj_vectors = structure_factor .* form_factors_all[ik] ./ sqrt(basis.model.unit_cell_volume)
-        projs[ik] = abs2.(ψk' * proj_vectors) # contract on p -> projs[ik][iband,m]
+       projs[ik] = abs2.(ψk' * projectors[ik])
+    end
+
+    return (;projs, labels)
+end
+
+"""
+This function extracts the required pdos from the output of the `compute_pdos` function. 
+
+    Input:
+     -> res         : Whole output from compute_pdos.
+     -> label (opt) : String for the whole orbital label. 
+     -> iatom (opt) : Atom number in the model.atoms vector. It becomes important for magnetically or spatially non-equivalent atoms. 
+    Output:
+     -> pdos        : Vector containing the pdos(ε).
+"""
+function get_pdos(pdos_res, projs::AbstractVector)
```

Comment:
Is there a need for this more complicated version of `get_pdos` (compared to `get_pdos(pdos_res; iatom=nothing, label=nothing)`)? The handling of the `projs` is a bit confusing.